### DISCUSSION TITLE
Airflow hooks communication

### DISCUSSION THREAD

**Discussion by albertcuy** [apache/airflow#32900] [Q&A]
Hi, just curious about the concept.

By definition, a server pushes out notifications/data to a client, but then(at least to me) it seems like the Airflow S3Hook is used much more like an API -- i.e. the client running airflow polls the server when it executes .download()

Can someone set me straight on this? Thanks :)

**Comment by potiuk** [comment]
Not sure where definition of server is from. I know quite the opposite definition of server/client. Notification from server are usually only possible when you do async communication, but for synchronous communication (which is still vast majority of usages of client/server ) client requests something and server provides it.
